integrations/google_genai/src/haystack_integrations/components/embedders/google_genai/document_embedder.py

```diff
@@ -77,16 +119,25 @@ def __init__(
             For more information, see the [Google AI Task types](https://ai.google.dev/gemini-api/docs/embeddings#task-types).
         """
         self._api_key = api_key
+        self._use_vertex_ai = use_vertex_ai
```

Comment:
same suggestion here as above regarding the replacement of the bool, this function would then call the `_get_client()`